skyportal/handlers/api/catalog_services.py

```diff
@@ -460,3 +471,206 @@ def fetch_swift_transients(instrument_id, user_id):
+        start_date = data.get('startDate', None)
+        end_date = data.get('endDate', None)
+
+        if start_date is not None:
+            start_date = Time(arrow.get(start_date.strip()).datetime)
+        if end_date is not None:
+            end_date = Time(arrow.get(end_date.strip()).datetime)
+
+        payload = {'start_date': start_date, 'end_date': end_date}
+
+        with self.Session() as session:
+            telescope = session.scalars(
+                Telescope.select(session.user_or_token).where(
+                    Telescope.nickname == telescope_name
+                )
+            ).first()
+            if telescope is None:
+                return self.error(f'Expected a Telescope named {telescope_name}')
+            instrument = telescope.instruments[0]
+
+            fetch_tr = functools.partial(
+                fetch_gaia_transients,
+                instrument.id,
+                self.associated_user_object.id,
+                payload,
+            )
+
+            IOLoop.current().run_in_executor(None, fetch_tr)
+
+            return self.success()
+
+
+def fetch_gaia_transients(instrument_id, user_id, payload):
+    """Fetch Gaia Photometric Alert transients.
+    instrument_id : int
+        ID of the instrument
+    user_id : int
+        ID of the User
+    payload : dict
+        Dictionary containing filtering parameters
+    """
+
+    session = Session()
+    obj_ids = []
+
+    lightcurve_url = "https://gsaweb.ast.cam.ac.uk/alerts/alert"
+    alert_url = "http://gsaweb.ast.cam.ac.uk/alerts/alerts.csv"
+    # alert_url = "./alerts.csv"
+
+    try:
+        file_read = False
+        nretries = 0
+
+        while not file_read and nretries < 10:
+            try:
+                table = Table.read(alert_url, format='csv')
+                file_read = True
+            except FileNotFoundError:
+                nretries = nretries + 1
+                time.sleep(10)
+        if not file_read:
+            log('Failed to read Gaia catalog')
+            return
+
+        start_date = payload.get('start_date', None)
+        end_date = payload.get('end_date', None)
+
+        user = session.scalar(sa.select(User).where(User.id == user_id))
+
+        group_ids = [g.id for g in user.accessible_groups]
+        for row in table:
+            name = row['#Name']
+            ra, dec = row['RaDeg'], row['DecDeg']
+            date = Time(row['Date'], format='iso')
+
+            if start_date is not None:
+                if date < start_date:
+                    continue
+            if end_date is not None:
+                if date > end_date:
+                    continue
+
+            data = {'ra': ra, 'dec': dec, 'id': name}
+            s = session.scalars(Obj.select(user).where(Obj.id == name)).first()
+            if s is None:
+                obj_id = post_source(data, user_id, session)
+                obj_ids.append(obj_id)
+            else:
+                obj_id = s.id
```

Comment:
Are you sure you don't want to append to `obj_ids` in case where the source is found?

Reply:
I return those obj_ids that get posted to display on the front-end that they were added.